geode-redis/src/distributedTest/java/org/apache/geode/redis/PubSubDUnitTest.java

```diff
@@ -88,4 +91,60 @@ public void testSubscribePublishUsingDifferentServers() throws Exception {
     GeodeAwaitility.await().untilAsserted(subscriberThread2::join);
   }
 
+  @Test
+  public void testConcurrentPubSub() throws Exception {
+    Jedis subscriber1 = new Jedis("localhost", ports[0]);
+    Jedis subscriber2 = new Jedis("localhost", ports[1]);
+
+    CountDownLatch latch = new CountDownLatch(2);
+    MockSubscriber mockSubscriber1 = new MockSubscriber(latch);
+    MockSubscriber mockSubscriber2 = new MockSubscriber(latch);
+
+    Runnable runnable1 = () -> subscriber1.subscribe(mockSubscriber1, CHANNEL_NAME);
+    Thread subscriberThread1 = new Thread(runnable1);
+    subscriberThread1.start();
+
+    Runnable runnable2 = () -> subscriber2.subscribe(mockSubscriber2, CHANNEL_NAME);
+    Thread subscriberThread2 = new Thread(runnable2);
+    subscriberThread2.start();
+
+    assertThat(latch.await(2, TimeUnit.SECONDS))
+        .as("channel subscription was not received")
+        .isTrue();
+
+    int CLIENT_COUNT = 10;
+    int ITERATIONS = 1000;
+
+    List<Jedis> clients = new ArrayList<>();
+    List<Thread> threads = new ArrayList<>();
```

Comment:
Why not have an ExecutorPool rather than a specific list of Threads...

Reply:
No particular reason. Is there any advantage of one over the other in such a simple situation? I can change it.

Reply:
So, in short, seeing such a raw implementation of thread management is not something that I think we need to have. ExecutorPools are geared at managing threads and their lifecycles. Whilst I agree that this is a "simple" situation, I also agree that it is far easier to just have a payload(s) assigned to the an ExecutorPool.
Maybe not a gating factor.

Reply:
Switched to using `ExecutorServiceRule`